src/metabase/api/search.clj

```diff
@@ -286,52 +281,28 @@
       (update :select (fn [columns]
                         (cons [(hx/literal "dataset") :model] (rest columns))))))
 
-(defn- shared-collection-impl
-  [model search-ctx]
+(s/defmethod search-query-for-model "collection"
+  [_model search-ctx :- SearchContext]
   (-> (base-query-for-model "collection" search-ctx)
-      (update :where (fn [where] [:and [(if (= model "app") :<> :=) :app.id nil] where]))
       (hh/left-join [CollectionBookmark :bookmark]
                     [:and
                      [:= :bookmark.collection_id :collection.id]
-                     [:= :bookmark.user_id api/*current-user-id*]]
-                    [App :app]
-                    [:= :app.collection_id :collection.id])
+                     [:= :bookmark.user_id api/*current-user-id*]])
       (add-collection-join-and-where-clauses :collection.id search-ctx)))
```

Comment:
This was shared between apps and collections (see the `defmethod`s below), now it's collection only.